### DISCUSSION TITLE
How to change 250 character limit for xtts v2 model?

### DISCUSSION THREAD

**Discussion by fivestones** [coqui-ai/TTS#3197] [General]
I'm on macos with an M2 chip, installed tts with pip. It's working well but if I try to use a sentence with more than 250 characters I get a warning that audio will be truncated and it is indeed truncated. I've seen a couple of issues about adding a max_decoder_steps option in config.json (see https://github.com/coqui-ai/TTS/issues/1680 and https://github.com/coqui-ai/TTS/issues/1522) but I can't find such a file.

-Can this be done with xtts v2?
-If so, where might I find the config.json file for this model?

I did find a file ~/mambaforge/lib/python3.10/site-packages/TTS/.models.json which references a config.json URL (https://coqui.gateway.scarf.sh/hf-coqui/XTTS-v2/main/config.json) in the xtts_v2 section, but I'm not sure how to make use of that. I'm guessing it might download it? But then where does it go?

**Comment by genglinxiao** [comment]
See my comment here: https://github.com/coqui-ai/TTS/issues/1680#issuecomment-1812220966.
I changed the limit in TTS/tts/layers/xtts/tokenizer.py, and it works fine for now.

**Comment by FurkanGozukara** [comment]
> See my comment here: [#1680 (comment)](https://github.com/coqui-ai/TTS/issues/1680#issuecomment-1812220966). I changed the limit in TTS/tts/layers/xtts/tokenizer.py, and it works fine for now.

how many seconds you are generating with 250 characters?

I am trying to determine fine tuning longest length speech

**Reply by genglinxiao** [reply]
Here's one of the result I got on my M2 Macbook Air after I changed the limit to 2500.

['This blow settled Castor’s fate.', 'The account that Tiberius heard from Sejanus, who came to him with the mark of Castor’s slap still red on his cheek, was that Castor had been drunk when Sejanus had congratulated him on his protectorship and had struck him across the face saying: ‘Yes, it’s good to feel that I can do this now without fear of being hit back. And you can tell my father that I’ll do the same to every other dirty spy of his.’ Livilla confirmed this the next day when she came to complain that Castor had beaten her; she said that he had beaten her because she told him how disgusted she was with him for striking a man who could not strike back and for insulting his father.', 'Tiberius believed them.', 'He said nothing to Castor but put up a bronze statue of Sejanus in the theatre of Pompey, an extraordinary honour to be paid to any man in his lifetime.', 'This was understood to mean that Castor was out of favour with Tiberius in spite of his protectorship (for Sejanus and Livilla had circulated their version of the quarrel) and that Sejanus was now the one person whose favour was worth courting.', 'Many replicas of the statue were therefore made, which his partisans put in a place of honour in their halls on the right hand of Tiberius’s statue: but statues of Castor were rarely seen.', 'Castor’s face showed his resentment so clearly now whenever he met his father that Sejanus’s task was made easy.', 'He told Tiberius that Castor was sounding various senators as to their willingness to support him if he usurped the monarchy and that some of them had already promised their help.', 'The ones who seemed most dangerous to Tiberius were therefore arrested on the familiar charge of blaspheming against Augustus.', 'One man was condemned to death for having gone into a privy with a gold coin of Augustus’s in his hand.', 'Another was accused of having included a statue of Augustus in a list of furniture for sale in a country villa.', 'He would have been condemned to death if the Consul who was judging the case had not asked Tiberius to give his vote first.', 'Tiberius was ashamed to vote for the death-penalty, so the man was acquitted, but condemned soon after on another charge']
 > Processing time: 249.23891377449036
 > Real-time factor: 1.8772298538334593

I repeated this several times. I don't have a speed issue. However, the audio file generated may have issues from time to time. Sometimes it was several words missing, sometimes there were strange sound in between. Sometimes the results are perfectly fine.

**Reply by FurkanGozukara** [reply]
@genglinxiao i mean the length of the speech. like longest sentence is taking like 15 seconds. I am coding a web app like below

it is generating each speech and merging auto

it took 55 seconds on my case by the way

![screencapture-127-0-0-1-7860-2023-11-16-11_11_42](https://github.com/coqui-ai/TTS/assets/19240467/e4e83aba-c1fd-4d18-b63e-4ac47120dae3)


https://github.com/coqui-ai/TTS/assets/19240467/4b5e719f-6f37-469c-bc3e-738f19c4c01d

**Reply by RebeccaDBlack** [reply]
is this donald?

**Reply by FurkanGozukara** [reply]
yes it is

**Comment by erogol** [comment]
The next version will introduce infinite streaming. It does splitting the text automatically and lets you synthesize without cuts. I'd not help for basic inference runs but you can stream infinitely. Thanks to @WeberJulian

**Reply by FurkanGozukara** [reply]
so we will give whole text and it will generate a single file out of it? like 10 min 20 min etc

also what duration of training data do you suggest? like between 1.5 second to 15 seconds?

**Reply by Tillerz** [reply]
Is there a way to "pipe in" the text? So instead of `tts --text "test here"` something like `tts < input.txt`?

**Reply by fivestones** [reply]
> Is there a way to "pipe in" the text?

I've been using `--text “$(< text_you_want_to_read.txt)”` (for Linux) or `--text “$(cat text_you_want_to_read.txt)` (for macOS). Works pretty well, haven't had trouble so far.

**Comment by barghavanii** [comment]
Hi How I can train Xtts_v2 in unsupported dataset I feed my dataset in Persian but give me an error of zero length I guess because it is not supporting "fa" now how I can define new characters and language

**Reply by Tessory** [reply]
@barghavanii I also got error with zero length of my dataset for Russian language. But after addied "eval_split_size = 10," in config section of train_gpt_xtts.py , that error disappear. Did you train via train_gpt_xtts.py ? 
So just find line 
        save_checkpoints=True,
and add after that
        eval_split_size = 10,
That really helped me to run fine-turning training. But don't know how it's works.

**Reply by barghavanii** [reply]
thank you by this help you also can totally change vocab.json and replaced with your character ? @erogol @Edresson can you please guide me how I can replace my character (Persian ) with Arabic ? I can only replace a very few of them and when I applied  more changes it gave me 0 length error as attached you can find my vocab.json 
[vocab (11) (2).json](https://github.com/coqui-ai/TTS/files/14285590/vocab.11.2.json) I can't expand it more because I am finetuning the model but why I can't replace Arabic with my character ? @Tessory  If you also know please guide me
